apisix/plugins/consumer-restriction.lua

```diff
@@ -67,26 +81,33 @@ function _M.check_schema(conf)
     return true
 end
 
+
 function _M.access(conf, ctx)
-    if not ctx.consumer then
+    if not conf.type then
         return 401, { message = "Missing authentication or identity verification." }
     end
 
+    local type_value = type_funcs[conf.type](ctx)
+    if not type_value then
+        return 401, { message = "Missing `type` value." }
```

Comment:
"Missing `type` value." 
-> 
"failed to fetch value by value type: " .. conf.type